homeassistant/components/withings/config_flow.py

```diff
@@ -46,33 +48,60 @@ async def async_oauth_create_entry(self, data: dict) -> dict:
 
     async def async_step_profile(self, data: dict) -> dict:
         """Prompt the user to select a user profile."""
-        profile = data.get(const.PROFILE)
+        errors = {}
+        # pylint: disable=no-member # https://github.com/PyCQA/pylint/issues/3167
+        context_profile = self.context.get(const.PROFILE)
+        profile = data.get(const.PROFILE) or context_profile
 
         if profile:
-            new_data = {**self._current_data, **{const.PROFILE: profile}}
-            self._current_data = None
-            return await self.async_step_finish(new_data)
+            existing_entry = next(
+                iter(
+                    config_entry
+                    for config_entry in self.hass.config_entries.async_entries(
+                        const.DOMAIN
+                    )
+                    if slugify(config_entry.data.get(const.PROFILE)) == slugify(profile)
+                ),
+                None,
+            )
+
+            if context_profile or not existing_entry:
+                new_data = {**self._current_data, **{const.PROFILE: profile}}
```

Comment:
We don't need to create a single item dict that we unpack into the new dict. Just set that item in the dict.

```py
new_data = {**self._current_data, const.PROFILE: profile}
```